trading_calendars/calendar_helpers.py

```diff
@@ -25,14 +28,31 @@ def previous_divider_idx(dividers, minute_val):
     return divider_idx - 1
 
 
-def is_open(opens, closes, minute_val):
+def is_open(opens, break_starts, break_ends, closes, minute_val):
 
     open_idx = np.searchsorted(opens, minute_val)
     close_idx = np.searchsorted(closes, minute_val)
 
     if open_idx != close_idx:
-        # if the indices are not same, that means the market is open
-        return True
+        # we are not guarenteed to have a sorted list of breaks, since break
+        # may not exist for a given date. Thus we need special logic.
+
+        # if the indices are not same, that means we are within a session
+        if (break_starts == NP_NAT).all() and (break_ends == NP_NAT).all():
+            # this calendar has no breaks
+            return True
+
+        break_start_on_open_dt = break_starts[open_idx - 1]
+        break_end_on_open_dt = break_ends[open_idx - 1]
+        if break_start_on_open_dt == NP_NAT:
+            # There is no break on the relevant day
+            return True
+        elif break_start_on_open_dt <= minute_val <= break_end_on_open_dt:
+            # we're in the middle of a break
+            return False
+        else:
+            return True
+
     else:
         try:
             # if they are the same, it might be the first minute of a
```

Comment:
What is this case? Does it apply to the re-open minute after the break?

Reply:
This is now handled in the check above:

```
if break_start_on_open_dt <= minute_val < break_end_on_open_dt:
            # we're in the middle of a break
            return False
        else:
            return True
```

If `minute_val==break_end_on_open_dt` we'll return `True`.